libcaliptra/examples/generic/test.c

```diff
@@ -925,6 +925,161 @@ int sign_with_exported_ecdsa_cdi(const test_info* info)
+    };
+
+    struct caliptra_sign_with_exported_ecdsa_req sign_req = { 0 };
+    memcpy(&sign_req.exported_cdi_handle, exported_resp.exported_cdi_handle, sizeof(uint8_t) * 32);
+    memcpy(&sign_req.tbs, &tbs, sizeof(uint8_t) * 48);
+
+    struct caliptra_sign_with_exported_ecdsa_resp sign_resp = { 0 };
+    status = caliptra_sign_with_exported_ecdsa(&sign_req, &sign_resp, false);
+
+    if (status) {
+        printf("Sign with exported Ecdsa failed: 0x%x\n", status);
+        return 1;
+    }
+
+    if (caliptra_verify_ecdsa_signature(&exported_resp, &sign_resp, tbs, sizeof(tbs))) {
+        printf("Sign with exported Ecdsa: OK\n");
+    } else {
+        printf("Error invalid signature.\n");
+        failure = 1;
+    }
+
+    // Add new measurement.
+    memset(&derive_context_cmd, 0, sizeof(struct dpe_derive_context_cmd));
+    derive_context_cmd.cmd_hdr.magic = DPE_MAGIC;
+    derive_context_cmd.cmd_hdr.cmd_id = DPE_DERIVE_CONTEXT;
+    derive_context_cmd.cmd_hdr.profile = 0x4;
+    derive_context_cmd.flags = DPE_DERIVE_CONTEXT_FLAG_RECURSIVE;
+
+    memset(&dpe_req, 0, sizeof(struct caliptra_invoke_dpe_req));
+    dpe_req.derive_context_cmd = derive_context_cmd;
+    dpe_req.data_size = sizeof(struct dpe_derive_context_cmd);
+
+    memset(&dpe_resp, 0, sizeof(struct caliptra_invoke_dpe_resp));
+    status = caliptra_invoke_dpe_command(&dpe_req, &dpe_resp, false);
+
+    if (status) {
+        printf("DPE Command failed: 0x%x\n", status);
+        dump_caliptra_error_codes();
+        failure = 1;
+    } else {
+        printf("DPE Command: OK\n");
+    }
+
+    struct dpe_derive_context_response *derive_resp = &dpe_resp.derive_context_resp;
+
+    // Revoke existing exported CDI 
+    struct caliptra_revoke_exported_cdi_handle_req revoke_req = { 0 };
+    memcpy(&revoke_req.exported_cdi_handle, exported_resp.exported_cdi_handle, sizeof(uint8_t) * 32);
+
+    if (caliptra_revoke_exported_cdi_handle(&revoke_req, false)) {
+        printf("Error Revoke Exported CDI Handle.\n");
+        failure = 1;
+    } else {
+        printf("Revoke Exported CDI Handle: OK\n");
+    }
+
+    // Create new exported cdi handle and certificate
+    memset(&derive_context_cmd, 0, sizeof(struct dpe_derive_context_cmd));
+    derive_context_cmd.cmd_hdr.magic = DPE_MAGIC;
+    derive_context_cmd.cmd_hdr.cmd_id = DPE_DERIVE_CONTEXT;
+    derive_context_cmd.cmd_hdr.profile = 0x4;
+    derive_context_cmd.flags = DPE_DERIVE_CONTEXT_FLAG_EXPORT_CDI | DPE_DERIVE_CONTEXT_FLAG_CREATE_CERTIFICATE;
+    memcpy(derive_context_cmd.context_handle, derive_resp->new_context_handle, sizeof(uint8_t) * 16);
+
+    memset(&dpe_req, 0, sizeof(struct caliptra_invoke_dpe_req));
+    dpe_req.derive_context_cmd = derive_context_cmd;
+    dpe_req.data_size = sizeof(struct dpe_derive_context_cmd);
+
+    memset(&dpe_resp, 0, sizeof(struct caliptra_invoke_dpe_resp));
+    status = caliptra_invoke_dpe_command(&dpe_req, &dpe_resp, false);
```

Comment:
nit: since this is fairly verbose and done twice in this test, would it make sense to put into a helper function?

Reply:
Wrapped most of this with macros, so it's somewhat less verbose